src/api/middlewares/FusionAuthMiddleware.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response, NextFunction } from 'express'
+import { UNAUTHORIZED } from 'http-status-codes'
+import { ExpressMiddlewareInterface, Middleware, HttpError } from 'routing-controllers'
+import Container from 'typedi'
+
+import envs from '@/config/envs'
+import { UserRegistration, FusionAuthClient } from '@fusionauth/typescript-client'
+
+import { FusionAuthService } from '~/services'
+
+@Middleware({ type: 'before' })
+export default class FusionAuthMiddleware implements ExpressMiddlewareInterface {
+  private EXCLUDED_PATHS = ['/api/v1/status']
+
+  public fusionAuthClient: FusionAuthClient
+
+  constructor() {
+    this.fusionAuthClient = Container.get(FusionAuthService).client
+  }
+
+  public async use(req: Request, res: Response, next: NextFunction) {
+    if (this.EXCLUDED_PATHS.includes(req.path)) return next()
+
+    const {
+      response: {
+        user: { registrations }
+      }
+    } = await this.fusionAuthClient.retrieveUserUsingJWT(req.headers.authorization)
+    if (this.verifyUserRolePermission(registrations)) return next()
+
+    throw new HttpError(UNAUTHORIZED, 'Unauthorized')
```

Comment:
Isso veio daquela discussão sobre status 401/403
```suggestion
    throw new HttpError(FORBIDDEN, 'Forbidden')
```